plugin/src/init.server.lua

```diff
@@ -1,222 +1,278 @@
 		Headers = {
 			["Content-Type"] = "application/json",
 		},
 		Body = HttpService:JSONEncode({
-			tree = tree,
+			tree = Tree,
 		}),
+
 		Compress = Enum.HttpCompression.Gzip,
 	})
 
-	if not success then
-		warn(`[Luau Language Server] Connecting to server failed: {result}`)
-		cleanup()
-	elseif not result.Success then
-		warn(`[Luau Language Server] Sending full DM info failed: {result.StatusCode}: {result.Body}`)
-		cleanup()
+	if not Success then
+		warn(`[Luau Language Server] Connecting to server failed: {Result}`)
+		CleanUpConnections()
+	elseif not Result.Success then
+		warn(`[Luau Language Server] Sending DataModel info failed: {Result.StatusCode}: {Result.Body}`)
+		CleanUpConnections()
 	else
-		connected.Value = true
-		if not isSilent then
-			print(`[Luau Language Server] Successfully sent full DataModel info`)
-		end
+		Connected = true
+		ConnectButton.Icon = "rbxassetid://11116536087"
+		warn("[Luau Language Server] Now sending DataModel information.")
 	end
 end
 
-local function watchChanges(isSilent)
-	local sendTask: thread?
-	if connected.Value or Settings == nil then
-		if not isSilent then
-			warn("[Luau Language Server] Connecting to server failed: invalid settings")
-		end
+local function WatchChanges()
+	local SendTask: thread?
+	if Connected or Settings == nil then
+		warn("[Luau Language Server] Connecting to server failed: Already connected, or settings non-existent.")
 		return
 	end
-	cleanup()
+	CleanUpConnections()
 
-	local function deferSend()
-		if sendTask then
-			task.cancel(sendTask)
+	local function DeferSendDataModelInfo()
+		if SendTask then
+			task.cancel(SendTask)
 		end
-		sendTask = task.delay(0.5, function()
-			sendFullDMInfo(isSilent)
-			sendTask = nil
+		SendTask = task.delay(0.5, function()
+			SendDataModelInfo()
+			SendTask = nil
 		end)
 	end
 
-	-- TODO: we should only send delta info if possible
-	local function descendantChanged(instance: Instance)
-		for _, service in Settings.include do
-			if instance:IsDescendantOf(service) then
-				deferSend()
-				return
+	for _, Service: Instance in Settings.Include do
+		table.insert(
+			Connections,
+			Service.DescendantAdded:Connect(function(Descendant: Instance)
+				if table.find(CLASSES_EXCEPTION_LIST, Descendant.ClassName) ~= nil then
+					return
+				end
+				--print(`[Luau Language Server] {Descendant.Name} added to {Service.ClassName}`)
+				table.insert(
+					Connections,
+					Descendant.Changed:Connect(function(Property: string)
+						--print(`[Luau Language Server] {Property} changed on {Descendant.Name}`) Re enable for Verbose of some sort
+						DeferSendDataModelInfo()
+					end)
+				)
+			end)
+		)
+		for _, Descendant in Service:GetDescendants() do
+			if table.find(CLASSES_EXCEPTION_LIST, Descendant.ClassName) ~= nil then
+				continue
 			end
+			table.insert(
+				Connections,
+				Descendant.Changed:Connect(function(Property: string)
```

Comment:
`GetPropertyChangedSignal("Name")` here too

Reply:
Addressed in another comment